apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/http/HttpNetworkTransport.kt

```diff
@@ -67,32 +76,45 @@ private constructor(
         )
       }
 
-      // do not capture request
-      val operation = request.operation
-      val response = worker.doWork {
-        try {
-          operation.parseJsonResponse(
-              jsonReader = httpResponse.body!!.jsonReader(),
-              customScalarAdapters = customScalarAdapters
-          )
-        } catch (e: Exception) {
-          throw wrapThrowableIfNeeded(e)
-        }
+      // When using @defer, the response contains multiple parts, using the multipart content type.
+      // See https://github.com/graphql/graphql-over-http/blob/main/rfcs/IncrementalDelivery.md
+      val bodies: Flow<BufferedSource> = if (httpResponse.isMultipart) {
+        multipartBodyFlow(httpResponse)
+      } else {
+        flowOf(httpResponse.body!!)
       }
 
-      @Suppress("DEPRECATION")
-      emit(
-          response.newBuilder()
-              .requestUuid(request.requestUuid)
-              .addExecutionContext(
-                  HttpInfo(
-                      startMillis = millisStart,
-                      endMillis = currentTimeMillis(),
-                      statusCode = httpResponse.statusCode,
-                      headers = httpResponse.headers
-                  )
-              )
-              .build()
+      emitAll(
+          bodies.map { body ->
+            val operation = request.operation
+            // On native, body will be freezed in worker.doWork, but reading a BufferedSource is a mutating operation.
+            // So we read the bytes into a ByteString first, and create a new BufferedSource from it after freezing.
+            val bodyByteString = if (platform() == Platform.Native) body.readByteString() else null
```

Comment:
Here's a trick to not crash inside `worker.doWork` when reading the body on native. 

I'm not very satisfied with it, especially the explicit call to `platform()`... 

Performance-wise we're already dealing with bodies fully read in memory on Apple, so I think it's ok.

In any case LMK if you can think of a better way to deal with it 🙏

Reply:
Yea no silver bullet... That works for me 👍